code/modules/hivemind/mobs.dm

```diff
@@ -107,6 +107,17 @@
 
 	closet_interaction()
 
+	//if somebody put us into disposal unit, let's go out
+	if(istype(loc, /obj/machinery/disposal))
```

Comment:
why not just check if the loc is any kind of /obj?

Reply:
But some of them has a special code for such things. If we just change mob's loc isn't it cause issues with this object?